bin/wsl/diag.wsl2.sh

```
@@ -0,0 +1,391 @@
+            log "info" "$line"
+        done
+    fi
+}
+
+# Check for systemd-resolved (common in Ubuntu)
+function show_systemd_resolved() {
+    local not_available=false
+
+    if command -v resolvectl &>/dev/null; then
+        log "info" "\nNetwork-specific DNS configuration via systemd-resolved:"
+        local output=$(resolvectl status 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)